### PR TITLE
Convert EnvironmentIndicator component to Composition API

### PR DESCRIPTION
**Pull request by Mysterious-Dev**
All is in the title ^^.

Related to https://github.com/modrinth/omorphia/pull/100#discussion_r1375273793

### CODE REVIEW

File: lib/components/base/EnvironmentIndicator.vue [hunk 1]
```diff
@@ -49,47 +49,42 @@
...
+    default: false,
+  },
+  categories: {
+    type: Array,
+    required: false,
+    default() {
+      return []
     },
   },
 })
```
Comment (brawaru) [inline lib/components/base/EnvironmentIndicator.vue:90]: I think it's worth using a TypeScript generic here instead. Just `defineProps` wrapped with `withDefaults`.

**Review by brawaru — CHANGES_REQUESTED** [review]

File: lib/components/base/EnvironmentIndicator.vue (hunk 1 above)
Comment (Mysterious-Dev) [reply lib/components/base/EnvironmentIndicator.vue]: Done ^^.

File: lib/components/base/EnvironmentIndicator.vue [hunk 2]
```diff
@@ -47,52 +47,29 @@
...
-})
+withDefaults(
+  defineProps<{
+    type: string
+    serverSide?: string
+    clientSide?: string
+    typeOnly?: boolean
+    alwaysShow?: boolean
+    search?: boolean
+    categories?: Array<string>
```
Comment (brawaru) [inline lib/components/base/EnvironmentIndicator.vue:61]:
```suggestion
    categories?: string[]
```

Use shorter syntax for array types.

**Review by brawaru — CHANGES_REQUESTED** [review]
Just one small stylistic thing.

File: lib/components/base/EnvironmentIndicator.vue (hunk 2 above)
Comment (Mysterious-Dev) [reply lib/components/base/EnvironmentIndicator.vue]: Done ^^.

**Review by brawaru — APPROVED** [review]

**Review by triphora — APPROVED** [review]

**Comment by Geometrically** [comment]
pls fix conlicts

**Comment by Mysterious-Dev** [comment]
@Geometrically It's fixed ^^.

**Pull request merged**